### PR TITLE
Move versioned generator configs to API group folders

### PR DESCRIPTION
**Pull request by ulucinar**
### Description of your changes

Fixes #
Consumes https://github.com/aws-controllers-k8s/code-generator/pull/266 that implements support for specifying API versions of the generated CRDs.

Move `generator-config.yaml` files to unversioned API group parent folders, because an API Group may now contain multiple versions. Please also note that https://github.com/aws-controllers-k8s/code-generator/pull/266 introduces a backwards-incompatible behavior change: The default location of the generator configuration files, if one is not specified via the command-line with the `--` option, is now `apis/<API Group>/generator-config.yaml`.

When https://github.com/aws-controllers-k8s/code-generator/pull/266 is merged, I will update the consumed `code-generator` version accordingly.

I have:

- [x] Read and followed Crossplane's [contribution process].
- [ ] Run `make reviewable test` to ensure this PR is ready for review.

### How has this code been tested

I have tried running `provider-aws` with a modified configuration for the Cloudfront's `Distribution` resource that moves it to the version `v1beta1`.
[contribution process]: https://git.io/fj2m9

### CODE REVIEW

File: apis/cloudfront/generator-config.yaml [hunk 1]
```diff
@@ -1,3 +1,8 @@
+resources:
+  "Distribution":
+    api_versions:
+    - name: "v1alpha1"
```
Comment (ulucinar) [inline apis/cloudfront/generator-config.yaml:4]: Configuration example for the `Distribution` resource. I've tested generating `Distribution` at `v1beta1`.

**Comment by haarchri** [comment]
i wonder if we first can merge https://github.com/crossplane/provider-aws/pull/1027 and then update this PR

**Comment by muvaf** [comment]
@ulucinar This probably needs a rebase to show the correct diff as https://github.com/crossplane/provider-aws/pull/1027 is merged, which should cover most of the generated file diffs already.

**Comment by muvaf** [comment]
I get the following when I run `make reviewable` and it seems `check-diff` also fails because of that:
```
16:41:57 [ .. ] Generating apigatewayv2 controllers and CRDs
# sigs.k8s.io/json/internal/golang/encoding/json
../pkg/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
../pkg/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
make[3]: *** [Makefile:129: services] Error 1
make[2]: *** [Makefile:143: services.all] Error 2
make[1]: *** [build/makelib/common.mk:418: generate] Error 2
make: *** [build/makelib/common.mk:424: reviewable] Error 2
```

Do you get it in your local?

File: CODE_GENERATION.md [hunk 2]
```diff
@@ -46,9 +46,21 @@ ignore:
     - <other CRD kinds you want to ignore>
 ```
 
+If you'd like to specify an API version for a generated resource other than 
+the default `v1alpha1`, you can do so with a configiration in `generator-config.yaml`
```
Comment (muvaf) [inline CODE_GENERATION.md:50]:
```suggestion
the default `v1alpha1`, you can do so with a configuration in `generator-config.yaml`
```

**Review by muvaf — APPROVED** [review]
Thanks @ulucinar !

File: CODE_GENERATION.md (hunk 2 above)
Comment (ulucinar) [reply CODE_GENERATION.md]: Thanks @muvaf!

**Pull request merged**